### DISCUSSION TITLE
Upgrading from Laravel 5.8 to 6.0 fails - In PackageManifest.php line 122: Undefined index: name

### DISCUSSION THREAD

**Discussion by geldek** [laravel/framework#39676] [Q&A]
- Laravel Version: 5.8.13 - 6.0.4
- PHP Version: 7.4.16
- Database Driver & Version:

### Description:
When upgrading from version 5.8 to 6.0 "composer update" fails with the error message In PackageManifest.php line 122: Undefined index: name. When composer is downloading version 6.0 from git, it uses the following URL: https://api.github.com/repos/laravel/framework/zipball/372e0add8a58e0e3deb78f87cbb2dc40d3e0ff08 (from the file vendor/composer/installed.json). The problem with this zip file is that it contains  a PackageManifest.php file that is missing the following line: 

`$packages = $installed['packages'] ?? $installed;`

This line of code, however, is present in the repository https://github.com/laravel/framework/blob/6.x/src/Illuminate/Foundation/PackageManifest.php

### Steps To Reproduce:
`
composer create-project laravel/laravel="5.8.*" test
`
`
cd test
`

edit the file "composer.json" as follows (change the laravel framework version from 5.8.* to 6.0.*):
    "require": {
        "php": "^7.1.3",
        "fideloper/proxy": "^4.0",
        "laravel/framework": "6.0*",
        "laravel/tinker": "^1.0"
    },
`
composer update
`

Please update the file PackageManifest.php in the zip file.

**Comment by dneykov** [comment]
I have similar issue when installing Laravel 6 project on new machine using composer 2. Composer installs laravel 6.6.2 but `PackageManifest.php` is missing 

`$packages = $installed['packages'] ?? $installed;`

and then I get error on `php artisan package:discover`. If I downgrade to composer 1 it works. What are the options to use composer 2? Thanks.

EDIT: After posting this I discovered that Composer 2 is supported form Laravel 6.18.7 but my project is lock at Laravel 6.6.2 so no problem. I should use Composer 1
I think this also answers the question above.
